lib/matplotlib/backend_bases.py

```diff
@@ -3551,7 +3593,8 @@ def _key_press(self, event):
         name = self._keys.get(event.key, None)
         if name is None:
             return
-        self.tool_trigger_event(name, canvasevent=event)
+        try: self.tool_trigger_event(name, canvasevent=event)
```

Comment:
That should be in `tool.trigger(sender, canvasevent, data)` (line 3545) where we don't control things.
That's the only place where things could go wrong with the trigger of tools. And we should `try` and raise a warning if fails